### PR TITLE
C style guide updates

### PR DESCRIPTION
**Pull request by dgquintas**
Also cleaned up formatting (a couple of `code blocks` and some line breaks).

### CODE REVIEW

**Comment by jboeuf** [comment]
LGTM. Thanks!

File: doc/c-style-guide.md [hunk 1]
```diff
@@ -34,24 +35,34 @@ Header Files
 - All header files should have a #define guard to prevent multiple inclusion.
   To guarantee uniqueness they should be based on the file's path.
 
-  For public headers: include/grpc/grpc.h --> GRPC_GRPC_H
+  For public headers: `include/grpc/grpc.h` → `GRPC_GRPC_H`
```
Comment (nicolasnoble) [inline doc/c-style-guide.md]: Did your editor change --> into → or did you do that on purpose ?

Comment (dgquintas) [inline doc/c-style-guide.md]: copy-paste FTW!

**Comment by dgquintas** [comment]
@kpayson64 why the removal of the ready to merge label?

**Comment by dgquintas** [comment]
test this please

**Comment by kpayson64** [comment]
@dgquintas 
It looked like there was an outstanding comment about unicode characters in the .md file.  Is that not getting changed?

**Comment by dgquintas** [comment]
@kpayson64 I think @nicolasnoble was just curious and that the change is fine. @nicolasnoble can you confirm?

**Comment by dgquintas** [comment]
@nicolasnoble ping

**Comment by ctiller** [comment]
LGTM

**Pull request merged**